### DISCUSSION TITLE
Very basic question ...

### DISCUSSION THREAD

**Discussion by mikamyara** [mjhoptics/ray-optics#99] [Q&A]
I use the following basic script :
```python
%matplotlib widget
from rayoptics.environment import *

mainOpm = OpticalModel()

sm  = mainOpm['seq_model']
osp = mainOpm['optical_spec']
pm = mainOpm['parax_model']
em = mainOpm['ele_model']
pt = mainOpm['part_tree']
ar = mainOpm['analysis_results']

osp.pupil = PupilSpec(osp, key=['object', 'pupil'], value=20.0)
osp.field_of_view = FieldSpec(osp, key=['object', 'height'],flds=[2.0,])
sm.gaps[0].thi=20.0

mainOpm.add_thinlens(power=0.1)
sm.gaps[1].thi=50.0

sm.list_model()        

mainOpm.update_model()
layout_plt = plt.figure(FigureClass=InteractiveLayout, do_draw_rays=True, do_paraxial_layout=False, opt_model=mainOpm)
layout_plt.refresh()
```
I get the following picture :
![image](https://user-images.githubusercontent.com/91488188/194930980-db32c977-1064-4fd6-a816-cff7f82c37fe.png)

As you can see, I used a thin lens, which should be "perfect" without aberrations, etc I guess. However I do not understand why the central ray is not focusing at the same position as the two other beams. What do I wrong ?

Thanks a lot,
   Mike

**Comment by mjhoptics** [comment]
Good observation. This illustrates the fact that an optical system can only exhibit perfect imaging at best between a single pair of points. This is a consequence of the laws of geometrical optics and limits the performance of even "ideal" systems. So, no, you didn't do anything wrong.
By moving the object point off of the lens' axis, the image point becomes aberrated; we're probably seeing field curvature and astigmatism, as well as coma.
You can use the keyword argument do_draw_ray_fans on the [InteractiveLayout](https://ray-optics.readthedocs.io/en/latest/api/rayoptics.mpl.interactivelayout.html#rayoptics.mpl.interactivelayout.InteractiveLayout) to draw more rays in the lens layout.

**Comment by mikamyara** [comment]
Thanks a lot for your answer. I am sorry for such a basic question. I understand that what is called "Gauss conditions" was not clear to me. Now it is thanks to this simple numerical experiment using your peace of software. Thanks a lot !
  Mike
